tests/deploy/test_scheduler_agent.py

```diff
@@ -1139,8 +1141,8 @@ def make_resources(
     assert agent.executor_manager.executors["agent3"].execute_count == 0
     # verify that r2 is considered dirty now, despite being in an assumed good operational state
     assert agent.scheduler._state.resource_state[rid2] == state.ResourceState(
-        # we have no data indicating that operational state has deviated from desired state => still UP_TO_DATE
-        status=state.ResourceStatus.UP_TO_DATE,
+        # We are skipped, so not compliant
+        status=state.ComplianceStatus.NON_COMPLIANT,
```

Comment:
I have to consider this change some more. It's possible that it now deviates from the intent of the test, and that we should set up a different scenario instead (if this edge case is still possible that is).

Reply:
The scenario is still relevant, just slightly less of an edge case. But I think we should update the comment two lines up that says "despite being in an assumed good operational state".